lib/msal-common/src/cache/CacheHelpers.ts

```diff
@@ -56,114 +56,121 @@ export class CacheHelpers {
-        this.cacheStorage.removeItem(TemporaryCacheKeys.ORIGIN_URI);
+        await this.cacheStorage.removeItem(TemporaryCacheKeys.REQUEST_STATE);
+        await this.cacheStorage.removeItem(TemporaryCacheKeys.REQUEST_PARAMS);
+        await this.cacheStorage.removeItem(TemporaryCacheKeys.ORIGIN_URI);
     }
 
     /**
      * Get all access tokens in the cache
      * @param clientId
      * @param homeAccountIdentifier
      */
-    getAllAccessTokens(clientId: string, authority: string, resource?: string, homeAccountIdentifier?: string): Array<AccessTokenCacheItem> {
-        const results = this.cacheStorage.getKeys().reduce<Array<AccessTokenCacheItem>>((tokens, key) => {
-            const keyMatches = key.match(clientId) && key.match(authority) && key.match(resource) && key.match(homeAccountIdentifier);
-            if (keyMatches) {
-                const value = this.cacheStorage.getItem(key);
-                if (value) {
-                    try {
-                        const parseAtKey = JSON.parse(key) as AccessTokenKey;
-                        if (this.checkForExactKeyMatch(parseAtKey, clientId, authority, resource, homeAccountIdentifier)) {
-                            const newAccessTokenCacheItem = new AccessTokenCacheItem(parseAtKey, JSON.parse(value) as AccessTokenValue);
-                            return tokens.concat([ newAccessTokenCacheItem ]);
-                        }
-                    } catch (e) {
-                        throw ClientAuthError.createCacheParseError(key);
+    async getAllAccessTokens(clientId: string, authority: string, resource?: string, homeAccountIdentifier?: string): Promise<Array<AccessTokenCacheItem>> {
+        const cacheKeys: Array<string> = await this.cacheStorage.getKeys();
+        const tokenKeys: Array<string> = cacheKeys.filter((key) => {
+            return key.match(clientId) && key.match(authority) && key.match(resource) && key.match(homeAccountIdentifier);
+        });
+        const foundTokens: Array<Promise<AccessTokenCacheItem>> = tokenKeys.map(async (key) => {
+            const valueAtKey = await this.cacheStorage.getItem(key);
+            if (valueAtKey) {
+                try {
+                    const parsedAccessTokenKey: AccessTokenKey = JSON.parse(key) as AccessTokenKey;
+                    if (this.checkForExactKeyMatch(parsedAccessTokenKey, clientId, authority, resource, homeAccountIdentifier)) {
+                        const newAccessTokenCacheItem = new AccessTokenCacheItem(parsedAccessTokenKey, JSON.parse(valueAtKey) as AccessTokenValue);
+                        return newAccessTokenCacheItem;
                     }
+                } catch (e) {
+                    throw ClientAuthError.createCacheParseError(key);
                 }
             }
-            return tokens;
-        }, []);
+            return null;
+        });
 
-        return results;
+        return Promise.all(foundTokens).then(arr => arr.filter(item => item));
```

Comment:
This filter basically takes the resolution of all promises and then only returns Items that are not null resolutions;